pelican/urlwrappers.py

```diff
@@ -52,27 +54,30 @@ def as_dict(self):
     def __hash__(self):
         return hash(self.slug)
 
-    def _key(self):
-        return self.slug
-
     def _normalize_key(self, key):
         subs = self.settings.get('SLUG_SUBSTITUTIONS', ())
         return six.text_type(slugify(key, subs))
 
     def __eq__(self, other):
-        return self._key() == self._normalize_key(other)
+        if isinstance(other, URLWrapper):
```

Comment:
Perhaps this would be better as `if isinstance(other, self.__class__):` that way it will verify it's the correct type.

That way these will be different:

```python
Category('test') != Tag('test')
```